aries-site/src/examples/templates/filtering/FilteringTable/FilteringTable.js

```diff
@@ -0,0 +1,132 @@
+import { useContext } from 'react';
+import { Box, DataTable, Heading, Menu, ResponsiveContext } from 'grommet';
+import {
+  StatusWarningSmall,
+  StatusCriticalSmall,
+  StatusGoodSmall,
+  StatusUnknownSmall,
+} from 'grommet-icons';
+
+import {
+  FilterControls,
+  FiltersProvider,
+  useFilters,
+} from '../../FilterControls';
+import { servers } from './mockData';
+
+export const FilteringTable = () => {
+  // Define which attributes should be made available for the user
+  // to filter upon
+  const filtersConfig = [
+    {
+      property: 'hardware.health.summary',
+      label: 'Status',
+      filterType: 'CheckBoxGroup',
+    },
+    {
+      property: 'state.connected',
+      label: 'State',
+      filterType: 'CheckBoxGroup',
+      render: datum => (datum ? 'Connected' : 'Not Connected'),
+    },
+    {
+      property: 'hardware.powerState',
+      label: 'Power',
+      filterType: 'CheckBoxGroup',
+    },
+    {
+      property: 'hardware.model',
+      label: 'Model',
+      filterType: 'CheckBoxGroup',
+    },
+  ];
+
+  return (
+    <Box pad="small" gap="medium">
+      <Heading level={2} margin="none">
+        Servers
+      </Heading>
+      <FiltersProvider>
+        <Box gap="medium">
+          <Box align="start" direction="row" justify="between" wrap>
+            <FilterControls
+              data={servers}
+              filters={filtersConfig}
+              primaryKey="id"
+              searchFilter={{ placeholder: 'Search servers...' }}
+            />
+            <Menu
+              kind="toolbar"
+              label="Actions"
+              items={[{ label: 'Power On' }, { label: 'Power Off' }]}
+            />
+          </Box>
+          <ServerResults />
+        </Box>
+      </FiltersProvider>
+    </Box>
+  );
+};
+
+const columns = [
+  {
+    property: 'hardware.serialNumber',
+    header: 'Serial',
+  },
+  {
+    property: 'state.connected',
+    header: 'State',
+    render: datum => (datum.state.connected ? 'Connected' : 'Not Connected'),
+  },
+  {
+    property: 'hardware.powerState',
+    header: 'Power',
+  },
+  {
+    property: 'hardware.model',
+    header: 'Model',
+  },
+];
+
+const statusIcons = {
+  Warning: <StatusWarningSmall color="status-warning" size="small" />,
+  OK: <StatusGoodSmall color="status-ok" size="small" />,
+  Critical: <StatusCriticalSmall color="status-critical" size="small" />,
+  Unknown: <StatusUnknownSmall color="status-unknown" size="small" />,
+};
+
+const ServerResults = () => {
```

Comment:
Searching on columns other than name is not working. For example, if I wanted to search by server model.

![Screen Shot 2021-06-14 at 4 36 54 PM](https://user-images.githubusercontent.com/1756948/121968400-8268df00-cd2f-11eb-922c-27b7d75a4ec6.png)

Reply:
Give this a try:

Swap out L107-ish in FilterContext.js

```
if (searchTerm) {
      const searchString = searchTerm.toLowerCase();
      filterResults = filterResults.filter(
        item =>
          getTextFromJson(item)
            .toLowerCase()
            .indexOf(searchString) > -1,
      );
    }
```

AND

```
 // Recursively extract all text values from the json object and return
  // a concatenated string.
  const getTextFromJson = json => {
    const obj = {};

    if (json === null || json === undefined) {
      return '';
    }

    if (typeof json === 'string') {
      return json;
    }

    Object.keys(json).forEach(key => (obj[key] = getTextFromJson(json[key])));
    return Object.values(obj).join(' ');
  };
```

Reply:
👆 NOTE: This does not currently work for the boolean columns. Need to think about that... but could be outside of this PR.

Reply:
Woohoo, thanks! and yes I'm comfortable handling boolean separately.